### PR TITLE
Add erc20 validator && add flow & solana tokens

### PR DESCRIPTION
**Pull request by BillySh1**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "@metamask/contract-metadata": "^1.27.0",
     "@uniswap/token-lists": "^1.0.0-beta.24",
     "ajv": "^6.12.5",
+    "axios": "^0.24.0",
```
Comment (guanbinrui) [inline package.json:16]: Try to use `fetch`.

File: src/validator/erc20/validator.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+const localList = require("./erc20-list.json");
+const axios = require("axios");
+// const tlist = require("../../../dist/v0.0.21/1/tokens.json");
+
+const getErc20List = async () => {
```
Comment (guanbinrui) [inline src/validator/erc20/validator.js:5]:
```suggestion
const getERC20List = async () => {
```

File: src/validator/erc20/validator.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  await axios
+    .get("https://api.coingecko.com/api/v3/coins/list")
```
Comment (guanbinrui) [inline src/validator/erc20/validator.js:7]: Use CMC API. https://web-api.coinmarketcap.com/v1/cryptocurrency/map?aux=status,platform&listing_status=active,untracked&sort=cmc_rank

File: src/validator/erc20/validator.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+ * @param { Array<{name:string, id: string, symbol:string, address: string}> } tokenList
+ */
+const validator = async (tokenList) => {
+  // const erc20List = (await getErc20List()) || localList;
+  const erc20List = localList.data;
+  let res = 1;
+  const errArr = [];
+  for (let i = 0; i < tokenList.length; i++) {
+    const token = tokenList[i];
+    const isInList = erc20List.filter((item) => {
```
Comment (guanbinrui) [inline src/validator/erc20/validator.js:26]: Create a set first and then check the existence in that.

File: src/validator/erc20/validator.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        return (
+          token.address.toLowerCase() ==
+          item.platform.token_address.toLowerCase()
+        );
+      }
+    });
+    if (!isInList.length) {
+      errArr.push(token);
+      res = errArr;
+      // const ret = await validate(token);
```
Comment (guanbinrui) [inline src/validator/erc20/validator.js:37]: Remove unused code.

File: src/validator/erc20/validator.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    })
+    .catch((e) => {
+      console.log(e);
+    });
+};
+
+/**
+ * @param { Array<{name:string, id: string, symbol:string, address: string}> } tokenList
+ */
+const validator = async (tokenList) => {
```
Comment (guanbinrui) [inline src/validator/erc20/validator.js:19]: `validateTokenExistenceCMC`

File: src/validator/erc20/validator.js (hunk 5 above)
Comment (czv6803892) [reply src/validator/erc20/validator.js:37]: sure

File: package.json (hunk 1 above)
Comment (czv6803892) [reply package.json:16]: ok

File: src/validator/erc20/validator.js (hunk 6 above)
Comment (czv6803892) [reply src/validator/erc20/validator.js:19]: modified

File: src/validator/erc20/validator.js (hunk 3 above)
Comment (czv6803892) [reply src/validator/erc20/validator.js:7]: modified

File: src/validator/erc20/validator.js (hunk 2 above)
Comment (czv6803892) [reply src/validator/erc20/validator.js]: modified

File: src/validator/erc20/validator.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
+const localList = require("./erc20-list.json");
```
[the remark this answers is not in the document]
Comment (czv6803892) [reply src/validator/erc20/validator.js:1]: modified

File: src/validator/erc20/validator.js (hunk 4 above)
Comment (czv6803892) [reply src/validator/erc20/validator.js]: modified

File: scripts/generate-flow-token.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+const FlowTokens = require("../src/flow/flow.json");
+const { addChainId, generateTokenList } = require("./shared");
+
+const FlowMaskTokenList = generateTokenList([...addChainId(FlowTokens, 1)], {
+  name: "Mask",
+  logoURI:
+    "https://raw.githubusercontent.com/DimensionDev/Maskbook-Website/master/img/MB--CircleCanvas--WhiteOverBlue.svg",
```
Comment (NikoManiac) [inline scripts/generate-flow-token.js:7]: Why add mask token in flow?

**Pull request closed without merging**